### PR TITLE
Category pages

### PR DESCRIPTION
**Pull request by jonalport**
This replaces https://github.com/moneyadviceservice/frontend/pull/1098

I've cleaned out the errant commits, and incorporated the updates from the previous PR as well as our front-end review meeting.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -65,13 +65,9 @@ require(['common', 'jquery'], function(MAS, $) {
...
-        showIcon: true,
-        useButton: true,
-        triggerEl: '.category-detail__heading',
-        targetEl: '.category-detail__list-container',
-        targetItems: '.category-detail__list-item',
-        viewAllButton: '.js-category-detail__view-all',
-        numberItemsToDisplay: 6
+        triggerEl: '.js-category-detail__view-all',
+        targetEl: '.js-category-detail__list--extended'
+        // useButton: true
```
Comment (asmega) [inline app/assets/javascripts/application.js]: need this comment?

Comment (jonalport) [inline app/assets/javascripts/application.js]: No! :)

File: app/assets/stylesheets/components/page_specific/_link_list.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+.link-list__item:after {
+  content: ", ";
+}
+ 
+.link-list__item:last-child:after {
+  content: none;
+}
+
+.link-list__item__link {
+  color: $color-true-black !important;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/page_specific/_link_list.scss]: Can we use `:link` style states to negate need for !important here? (I can't remember if we spoke about this the other day?)

Comment (jonalport) [inline app/assets/stylesheets/components/page_specific/_link_list.scss]: Oh yes we did, well spotted

File: app/views/styleguide/pages_parent_category_page.html.erb [hunk 3]
```diff
@@ -1,16 +1,62 @@
...
+      <div class="l-category__top">
+        <div class="l-category__top__content">
+          <h1 aria-level="1" class="l-category__heading" role="heading">Debt and borrowing</h1>
+          <span class="l-category__top__background"></span>
+          <p class="l-category__top__intro">Before you sign up for a credit card, bank loan or store card, or add to an existing card or loan it makes sense to think about whether you really need to borrow money.</p>
+        </div>
+        <h2 aria-level="2" class="l-category__top__link-list-heading" role="heading">How can we help?</h2>
+        <ul class="link-list">
+            <li class="link-list__item">
+              <a class="link-list__item__link" href="/en/">Before you borrow</a>
```
Comment (benbarnett) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Too much indentation here I think

Comment (jonalport) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Correct

**Comment by benbarnett** [comment]
:+1: after those tiny comments

File: db/schema.rb [hunk 4]
```diff
@@ -77,20 +77,6 @@
     t.datetime "updated_at",                             null: false
   end
 
-  create_table "advice_plans_advice_plans", force: true do |t|
```
Comment (asmega) [inline db/schema.rb]: don't checkin these changes on this file as they are unrelated.

File: app/views/categories/_content_items.html.erb [hunk 5]
```diff
@@ -1,7 +1,15 @@
 <ol class="category-detail__list">
-  <% contents.each do |item| %>
+  <% contents.take(6).each do |item| %>
```
Comment (tomas-stefano) [inline app/views/categories/_content_items.html.erb]: Why take 6? Can we move this to a method on controller and move this magic number as constant?

File: app/views/categories/_content_items.html.erb [hunk 6]
```diff
@@ -1,7 +1,15 @@
...
     <li class="category-detail__list-item">
       <%= link_to item.title, item.path %>
     </li>
   <% end %>
 </ol>
+<ol class="collapsible-section category-detail__list category-detail__list--extended js-category-detail__list--extended">
+  <% (contents[6..-1] || []).each do |item| %>
```
Comment (tomas-stefano) [inline app/views/categories/_content_items.html.erb]: I would move this to a method on the helper or controller. (What you prefer).

File: app/decorators/category_decorator.rb [hunk 7]
```diff
@@ -21,12 +20,24 @@ def render_contents
...
+    object.images.present? && object.images['large'].present?
+  end
+
+  def small_image?
+    object.images.present? && object.images['small'].present?
+  end
+
+  def images?
+    large_image? && small_image?
+  end
```
Comment (tomas-stefano) [inline app/decorators/category_decorator.rb]: Is there specs for this methods?

File: db/schema.rb (hunk 4 above)
Comment (Guntrisoft) [inline db/schema.rb]: What's going on here with removal of table?

File: app/views/categories/show.html.erb [hunk 8]
```diff
@@ -7,11 +7,40 @@
...
+        only screen and (min--moz-device-pixel-ratio: 2),
+        only screen and (-o-min-device-pixel-ratio: #{(2*10)}/10),
+        only screen and (min-device-pixel-ratio: 2),
+        only screen and (min-resolution: #{(2*96)}dpi),
+        only screen and (min-resolution: #{2}dppx) {
+          .l-category--image {
+            background-image: url(<%= category.images['large'] %>);
+          }
+        }
+      </style>
```
Comment (neoeno) [inline app/views/categories/show.html.erb]: Is there a better place for this? And the SVGs? There's a lot of set-up and style going on here

File: app/views/styleguide/pages_parent_category_page.html.erb [hunk 9]
```diff
@@ -1,16 +1,62 @@
...
+      @media only screen and (-webkit-min-device-pixel-ratio: 2),
+      only screen and (min--moz-device-pixel-ratio: 2),
+      only screen and (-o-min-device-pixel-ratio: #{(2*10)}/10),
+      only screen and (min-device-pixel-ratio: 2),
+      only screen and (min-resolution: #{(2*96)}dpi),
+      only screen and (min-resolution: #{2}dppx) {
+        .l-category--image {
+          background-image: url(<%= image_url('atm_en.png') %>);
+        }
+      }
```
Comment (neoeno) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Same here — and are the SVG tags above the same as the SVG files?

File: app/views/styleguide/pages_parent_category_page.html.erb [hunk 10]
```diff
@@ -1,16 +1,62 @@
...
+            <a class="link-list__item__link" href="/en/">Mortgage and rent problems</a>
+          </li>
+          <li class="link-list__item">
+            <a class="link-list__item__link" href="/en/">Help with loans</a>
+          </li>
+          <li class="link-list__item">
+            <a class="link-list__item__link" href="/en/">Credit cards and credit ratings</a>
+          </li>
+        </ul>
+      </div>
```
Comment (neoeno) [inline app/views/styleguide/pages_parent_category_page.html.erb]: No Welsh I guess?

File: app/views/styleguide/shared/_category_detail.html.erb [hunk 11]
```diff
@@ -1,67 +1,161 @@
...
+<section class="category-detail" data-dough-component="Collapsable">
+  <h2 aria-level="2" class="category-detail__heading" data-dough-collapsable-trigger="category_before-you-borrow" id="before-you-borrow" role="heading"><svg xmlns="http://www.w3.org/2000/svg" class="svg-icon svg-icon--plus svg-icon--plus--yellow">
+    <use xlink:href="#svg-icon--plus"></use>
+  </svg>
 
-  <p>Hear what young people like you have to say about how they are making their
-    first financial decisions and how they are coping with the pressures and opportunities this brings.</p>
+  <svg xmlns="http://www.w3.org/2000/svg" class="svg-icon svg-icon--minus svg-icon--minus--yellow">
+    <use xlink:href="#svg-icon--minus"></use>
+  </svg>
```
Comment (neoeno) [inline app/views/styleguide/shared/_category_detail.html.erb]: I seem to remember a use_icon partial somewhere earlier in this diff?

File: app/assets/stylesheets/components/page_specific/_featured_links.scss [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    padding-left: $baseline-unit*6;
+  }
+}
+
+.featured-links__item {
+  @include column(6);
+
+  &:before {
+    // This is to override the highly specific .editorial list styling
+    content: none !important;
```
Comment (neoeno) [inline app/assets/stylesheets/components/page_specific/_featured_links.scss]: Is this a sign something else needs fixing? (No idea)

File: app/views/categories/_child_categories.html.erb [hunk 13]
```diff
@@ -1,17 +1,26 @@
...
+                    class: 'category-detail__heading',
+                    data: { dough_collapsable_trigger: 'category_' + child_category.id } do
+    %>
+      <%= render 'shared/svg/use_icon', icon: 'plus', variant: 'yellow' %>
+      <%= render 'shared/svg/use_icon', icon: 'minus', variant: 'yellow' %>
+      <%= child_category.title %>
+    <% end %>
     <p class="category-detail__intro"><%= child_category.description %></p>
-
+    <%= render 'category_promos', promos: child_category.category_promos unless !child_category.category_promos || child_category.category_promos.empty? %>
```
Comment (neoeno) [inline app/views/categories/_child_categories.html.erb]: This could probably be put more concisely, but without knowing what kind of object `child_category` is it's hard to say. Could be `render [...] if child_category.category_promos.present?`

File: app/views/categories/_content_items.html.erb (hunk 6 above)
Comment (neoeno) [inline app/views/categories/_content_items.html.erb]: What are you expecting `contents` to be in the case that it fails over to use the empty array?

Comment (jonalport) [inline app/views/categories/_content_items.html.erb]: That's just in case contents has < 6 items to begin with

File: app/views/styleguide/pages_parent_category_page.html.erb (hunk 10 above)
Comment (jonalport) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Just the style guide, so I think not?

File: app/views/categories/_content_items.html.erb (hunk 6 above)
Comment (neoeno) [inline app/views/categories/_content_items.html.erb]: Aha, that makes sense. In that case, you could put this another way:

```ruby
Array(contents[6..-1]).each do |item|
```

Up to you — I find the latter clearer but others might feel differently.

File: app/views/styleguide/pages_parent_category_page.html.erb (hunk 10 above)
Comment (neoeno) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Oops, totally missed that this was part of the styleguide. In that case many of these comments will be overly-picky I suspect. Sorry!

File: app/decorators/category_decorator.rb (hunk 7 above)
Comment (jonalport) [inline app/decorators/category_decorator.rb]: Phil is adding some in for us

File: app/views/categories/_content_items.html.erb (hunk 6 above)
Comment (jonalport) [inline app/views/categories/_content_items.html.erb]: Thanks, that does read better. I think we will extract this in to a method in this case.

File: app/views/styleguide/pages_parent_category_page.html.erb (hunk 10 above)
Comment (jonalport) [inline app/views/styleguide/pages_parent_category_page.html.erb]: Heh, no worries :)

**Comment by asmega** [comment]
:+1:
